Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -2410,11 +2411,67 @@ static IntRect computeMarqueeSelectionRect(const WebCore::IntPoint& point1, cons
     return true;
 }
 
-bool UnifiedPDFPlugin::performDictionaryLookupAtLocation(const FloatPoint&)
+bool UnifiedPDFPlugin::performDictionaryLookupAtLocation(const FloatPoint& rootViewPoint)
 {
+    IntPoint pluginPoint = convertFromRootViewToPlugin(roundedIntPoint(rootViewPoint));
+    IntPoint documentPoint = convertFromPluginToDocument(pluginPoint);
+    auto pageIndex = pageIndexForDocumentPoint(documentPoint);
+    if (!pageIndex)
+        return false;
+
+    auto page = m_documentLayout.pageAtIndex(*pageIndex);
+    RetainPtr lookupSelection = [page selectionForWordAtPoint:convertFromDocumentToPage(documentPoint, *pageIndex)];
+    return searchInDictionary(WTFMove(lookupSelection));
+}
+
+bool UnifiedPDFPlugin::searchInDictionary(const RetainPtr<PDFSelection>& lookupSelection)
+{
+    RetainPtr scaledString = [lookupSelection attributedStringScaled:(m_scaleFactor * m_documentLayout.scale())];
+
+    if (auto selectionBounds = selectionBoundsForFirstPageInDocumentSpace(lookupSelection))
+        return showDefinitionForAttributedString(WTFMove(scaledString), *selectionBounds);
+
     return false;
 }
 
+std::optional<IntRect> UnifiedPDFPlugin::selectionBoundsForFirstPageInDocumentSpace(const RetainPtr<PDFSelection>& selection) const
+{
+    if (!selection)
+        return std::nullopt;
+
+    for (PDFPage *page in [selection pages]) {
+        auto pageIndex = m_documentLayout.indexForPage(page);
+        if (!pageIndex)
+            continue;
+        auto rectForPage = IntRect { [selection boundsForPage:page] };
+        return convertFromPageToDocument(rectForPage, *pageIndex);
+    }
+
+    return std::nullopt;
+}
+
+bool UnifiedPDFPlugin::showDefinitionForAttributedString(RetainPtr<NSAttributedString>&& string, const IntRect& rectInDocumentSpace)
```

Comment:
I just realized that `PDFPlugin` has a function with a very similar signature. I should unify (ha!) them somehow, or at least delegate the "inform webpage that a dictionary lookup was performed" bit to `PDFPluginBase`.

Reply:
I'll do this in a future commit. Landing this as-is for now.